### PR TITLE
Uses API data to update 'product_values'

### PR DESCRIPTION
**Pull request by riteshshrv**
- Replaces hard-coded decimal values in 'product_values'

### CODE REVIEW

File: tests/json/products/product-1.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
                                                                               "URL": {"value": "http://g-ecx.images-amazon.com/images/G/01/x-site/icons/no-img-sm._V192198896_.gif"},
                                                                               "Width": {"Units": {"value": "pixels"},
                                                                                         "value": "60"}},
+                                                               "ListPrice": {"Amount": {"value":"50.00"}, "CurrencyCode":{"value":"USD"}},
                                                                "Title": {"value": "A red coffee mug"},
```
Comment (priyankarani) [inline tests/json/products/product-1.json]: You should not make changes in json files. These are exactly what returned from amazon

File: product.py [hunk 2]
```diff
@@ -122,8 +121,8 @@ def create_using_amazon_data(cls, product_data):
         product_values.update({
             'products': [('create', [{
                 'code': product_data['Id']['value'],
-                'list_price': Decimal('0.01'),
-                'cost_price': Decimal('0.01'),
+                'list_price': product_attributes['ListPrice']['Amount']['value'],  # noqa
```
Comment (tarunbhardwaj) [inline product.py]: only when "ListPrice" is available and Price will always be in currency.

```<ns2:ListPrice>
            <ns2:Amount>44.99</ns2:Amount>
            <ns2:CurrencyCode>USD</ns2:CurrencyCode>
          </ns2:ListPrice>```

File: product.py [hunk 3]
```diff
@@ -119,11 +120,20 @@ def create_using_amazon_data(cls, product_data):
             product_attributes
         )
 
+        if 'ListPrice' in product_attributes.keys():
```
Comment (tarunbhardwaj) [inline product.py]: `if product_attributes.get('ListPrice'):`

File: product.py [hunk 4]
```diff
@@ -119,11 +120,20 @@ def create_using_amazon_data(cls, product_data):
...
+            list_price = product_attributes['ListPrice']['Amount']['value']
+            currency_code = product_attributes['ListPrice']['CurrencyCode']['value']  # noqa
+            list_price = Currency.compute(
+                Currency(currency_code), float(list_price), cls.currency
```
Comment (priyankarani) [inline product.py]: fetch company from active channel and then fetch currency from that.

check extract_product_values_from_amazon_data method how to fetch channel

File: product.py [hunk 5]
```diff
@@ -119,11 +121,26 @@ def create_using_amazon_data(cls, product_data):
...
         )
 
+        amazon_channel = SaleChannel(
+            Transaction().context['current_channel']
+        )
+        assert amazon_channel.source == 'amazon_mws'
+        if product_attributes.get('ListPrice'):
+            list_price = product_attributes['ListPrice']['Amount']['value']
+            currency_code = product_attributes['ListPrice']['CurrencyCode']['value']  # noqa
+            currency_id = Currency.search([('code', '=', currency_code), ])
```
Comment (priyankarani) [inline product.py]: it returns active record, not ID.
just 
currency = Currency.search([('code', '=', currency_code), limit=1])

File: product.py [hunk 6]
```diff
@@ -119,11 +121,26 @@ def create_using_amazon_data(cls, product_data):
...
+            list_price = Currency.compute(
+                Currency(currency_id), float(list_price),
+                amazon_channel.currency
+            )
+        else:
```
Comment (priyankarani) [inline product.py]: no need

File: product.py [hunk 7]
```diff
@@ -119,11 +121,26 @@ def create_using_amazon_data(cls, product_data):
             product_attributes
         )
 
+        amazon_channel = SaleChannel(
+            Transaction().context['current_channel']
+        )
+        assert amazon_channel.source == 'amazon_mws'
```
Comment (priyankarani) [inline product.py]: list_price = Decimal('0.01'), just before IF.

File: product.py [hunk 8]
```diff
@@ -119,11 +121,26 @@ def create_using_amazon_data(cls, product_data):
...
+            )
+        else:
+            list_price = None
+
         product_values.update({
             'products': [('create', [{
                 'code': product_data['Id']['value'],
-                'list_price': Decimal('0.01'),
-                'cost_price': Decimal('0.01'),
+                'list_price': list_price or Decimal('0.01'),
```
Comment (priyankarani) [inline product.py]: just list_price no need for 'or'

File: product.py [hunk 9]
```diff
@@ -119,11 +121,26 @@ def create_using_amazon_data(cls, product_data):
...
+                'cost_price': list_price or Decimal('0.01'),
```
Comment (priyankarani) [inline product.py]: same

File: product.py [hunk 10]
```diff
@@ -119,11 +121,28 @@ def create_using_amazon_data(cls, product_data):
...
+        amazon_channel = SaleChannel(
+            Transaction().context['current_channel']
+        )
+        assert amazon_channel.source == 'amazon_mws'
+
+        list_price = Decimal('0.01')
+        if product_attributes.get('ListPrice'):
+            list_price = product_attributes['ListPrice']['Amount']['value']
+            currency_code = product_attributes['ListPrice']['CurrencyCode']['value']  # noqa
+            currency = Currency.search([
```
Comment (priyankarani) [inline product.py]: it returns list of active records. so you have to untuple list or take first record from list

**Comment by priyankarani** [comment]
LGTM

File: channel.py [hunk 11]
```diff
@@ -459,7 +459,7 @@ def export_product_prices(self):
 
         return len(pricing_xml)
 
-    def import_product(self, sku):
+    def import_product(self, sku, product_data=None):
```
Comment (priyankarani) [inline channel.py]: pass same arg to super call too

File: product.py [hunk 12]
```diff
@@ -119,11 +121,28 @@ def create_using_amazon_data(cls, product_data):
...
+        list_price = Decimal('0.01')
+        if product_attributes.get('ListPrice'):
+            list_price = product_attributes['ListPrice']['Amount']['value']
+            currency_code = product_attributes['ListPrice']['CurrencyCode']['value']  # noqa
+            currency, = Currency.search([
+                ('code', '=', currency_code),
+            ], limit=1)
+            list_price = Currency.compute(
+                currency, Decimal(list_price),
+                amazon_channel.currency
```
Comment (tarunbhardwaj) [inline product.py]: company's currency

**Pull request merged**